### PR TITLE
Rename hypervisor extension "operating mode" to "privilege mode"

### PR DESCRIPTION
**Pull request by jhauser-us**
For the hypervisor extension, define "privilege mode" as the combination of virtualization mode and the new concept of "nominal privilege mode".

### CODE REVIEW

**Comment by ethindp** [comment]
I'm not up to speed on RISC-v developments but I feel like the name is ambiguous. To me, "privilege mode" (or the past tense of the phrase) means "any mode that is capable of executing privileged system instructions". I feel like individualizing the modes makes it clearer as to what mode you are referring to, instead of calling any privileged execution mode or combination of modes a "privilege mode". Could you provide a rationale for this change? (I'm not sure what the "nominal privilege mode" does or does not allow compared to the "virtualization mode", so I might be missing something.)

**Comment by jhauser-us** [comment]
The RISC-V ISA manual Volume II makes clear that U (user) is considered one of the privilege levels, and U-mode is one of the privilege modes for a hart.  This has been the case from the beginning.  This pull request makes no change in that regard.

**Comment by ethindp** [comment]
I don't think I ever implied that it did. I am simply trying to understand why this PR is necessary.

**Comment by gfavor** [comment]
I'll bite.  Most people I interact with in RISC-V already say "privilege mode" when referring to what is currently called "operating mode.  That is conceptually natural and consistent with the idea or intuition that the H extension simply adds two more privilege modes to the existing three.  So why introduce a new term, i.e. operating modes.

The architecture already distinguishes (encoding-wise in various CSRs) between the Virtualization Mode and the Privilege Level. 
 (Btw, the current Priv spec notes that priv modes and levels are often tied together or equated with each other, but also explains the fine distinction between the two.)  So maintaining "privilege mode" as being the overall privilege mode (one of five) maintains greater consistency with the current Priv arch, and then using the term Nominal Privilege instead of Privilege Encoding is (in my mind) a good clarification or better way to refer to that.  Especially since this mirrors the idea - with this virtualization architecture (unlike ARM's) - that there are three "base" privilege levels/modes and then H introduces virtualized variants of two of those (i.e. VS and VU versus HS and U modes).

In short, at least from my own perspective, this change is a clearer expression of what is going on with privilege levels and  modes, and does so without introducing a new third term (operating modes).

**Comment by ethindp** [comment]
@gfavor I understand now, and now that I'm able to look at the diff (I wasn't able to earlier -- my screen reader was making it a bit difficult :P) I do see the clarification. Thanks for the explanation.

**Comment by kasanovic** [comment]
I think people are confusing "privilege mode" (what your current mode is) with "privileged mode" (one that has privileges).
